File: 코딩 기초 트레이닝/세 개의 구분자-p181862/세 개의 구분자.py
# https://school.programmers.co.kr/learn/courses/30/lessons/181862

def solution(myStr):
    answer = []
    sub = ''

    myStr = list(myStr)
    for s in myStr:
        if not s in ['a', 'b', 'c']:
            sub += s
        elif len(sub) > 0:
            answer.append(sub)
            sub = ''

    if len(sub) > 0 :
        answer.append(sub)

    return ['EMPTY'] if len(answer) == 0 else answer

File: 코딩 기초 트레이닝/세 개의 구분자-p181862/세 개의 구분자.js
// https://school.programmers.co.kr/learn/courses/30/lessons/181862

function solution(myStr) {
    const answer = [];
    const op = ['a','b','c'];

    let sub = '';

    myStr = myStr.split("");

    myStr.forEach((s) => {
        if(!op.includes(s)) {
            sub += s;
        } else if(sub.length > 0) {
            answer.push(sub);
            sub = '';
        }
    });

    if(sub.length > 0) answer.push(sub);


    return answer.length === 0 ? ['EMPTY'] : answer;
}
